macros/REST_Raw_PlotVetoData.C

```diff
@@ -40,13 +32,25 @@ Int_t REST_Raw_PlotVetoData(string fileName = "data/R01133_000_RawToSignal_Ar2Is
 	vector<string> groupNames;
 	vector<string> groupIds;
 	
-	// vector<int> nVeto; // number of vetoes in group
-
 	// check if signal IDs are defined as a single list or in groups
+	// if vetoes are defined in groups:
 	if (vetoSignalId[0] == -1.){
-		pair<vector<string>,vector<string>> vetoGroups = veto->GetVetoGroups();
-		groupNames = std::get<0>(vetoGroups);
-		groupIds   = std::get<1>(vetoGroups);
+		// check if all groups or only a selected group should be plotted
+		if (group == "all"){
+			pair<vector<string>,vector<string>> vetoGroups = veto->GetVetoGroups();
+			groupNames = std::get<0>(vetoGroups);
+			groupIds   = std::get<1>(vetoGroups);
+		} else {
+			groupNames.push_back(group);
+			groupIds.push_back(veto->GetGroupIds(group));
+			// In case group name doesn't exist:
+			if (std::strcmp(string(groupIds[0]).c_str(),string("-1").c_str())==0){
```

Comment:
I guess it does not need that complex expression.

a `if( groupIds[0] == "-1" )` should be enough. It is just a string comparison.

Reply:
I tried this, but it didn't work.